streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java

```diff
@@ -668,6 +671,9 @@ private void checkKeyValueStoreMetrics(final String group0100To24,
         checkMetricByName(listMetricStore, SUPPRESSION_BUFFER_SIZE_CURRENT, 0);
         checkMetricByName(listMetricStore, SUPPRESSION_BUFFER_SIZE_AVG, 0);
         checkMetricByName(listMetricStore, SUPPRESSION_BUFFER_SIZE_MAX, 0);
+        checkMetricByName(listMetricStore, RECORD_E2E_LATENCY_AVG, expectedNumberofE2ELatencyMetrics);
+        checkMetricByName(listMetricStore, RECORD_E2E_LATENCY_MIN, expectedNumberofE2ELatencyMetrics);
+        checkMetricByName(listMetricStore, RECORD_E2E_LATENCY_MAX, expectedNumberofE2ELatencyMetrics);
```

Comment:
Ok there's something I'm not understanding about this test and/or the built-in metrics version. For some reason, the KV-store metrics are 0 when `METRICS_0100_TO_24` is used, and 1 (as expected) when the latest version in used. I feel like this is wrong, and it should always be 1, but I need some clarify on how this config is supposed to be used
What makes me pretty sure there's something actually wrong here is that for the window/session store metrics, they are actually always at 1. But I can't figure out why the KV store metrics would be any different than the others. Any ideas @cadonna ?

Reply:
I agree with you, it should always be 1. It is the group of the metrics. See my comment in `StateStoreMetrics`. I am glad this test served its purpose, because I did not notice this in the unit tests!

Reply:
Sorry, I did a mistake here. We should not give new metrics old groups. I think to fix this test you need to adapt the filter on line 618 to let all metrics with groups that relate to KV state stores pass. See `checkWindowStoreAndSuppressionBufferMetrics()` for an example.

Reply:
Hm, if I let additional metrics through (by allowing all `STATE_STORE_LEVEL_GROUP` metrics regardless of buildInMetricsVersion) then the test fails much earlier. For example
```java.lang.AssertionError: Size of metrics of type:'put-latency-avg' must be equal to 1 but it's equal to 3 ```

I think I could just do the check for the e2e latency before filtering, but I'm worried there's something else wrong here. Why do we get 3 `put-latency-avg` metrics if we let all store metrics through in `checkKeyValueStoreMetrics` but not when we let all metrics through in `checkWindowStoreAndSuppressionBufferMetrics`?

Reply:
You can use the following to get it right without the need to do the check for the e2e latency before filtering

```
.filter(m -> m.metricName().tags().containsKey(tagKey) && 
    (m.metricName().group().equals(group0100To24) || m.metricName().group().equals(STATE_STORE_LEVEL_GROUP))
).collect(Collectors.toList());

```

The reason for the difference between the KV store and the window store is that they are used in different tests with different number of state stores.

The test that uses the KV stores tests three different types of KV stores, namely in-memory, rocksdb, and in-memory-lru-cache. For each of this types the old group name changes. That is also the reason we need to pass the parameter `group0100To24` to `checkKeyValueStoreMetrics()`.

In `checkWindowStoreAndSuppressionBufferMetrics()` we need to filter for four groups, because the corresponding test uses suppression and window state store. Suppression buffers had their own groups in the old version. In the new version they moved into the state store group. Those groups are `BUFFER_LEVEL_GROUP_0100_TO_24` and `STATE_STORE_LEVEL_GROUP`. The window state store had their own group in the old version, i.e., `STATE_STORE_LEVEL_GROUP_ROCKSDB_WINDOW_STORE_0100_TO_24` (we are only using RocksDB-based window stores in the test). Finally, during the implementationof KIP-444, we discovered that we named a group incorrectly. That why we filter also for group `stream-rocksdb-window-metrics`.

So to sum up, it is hard to compare the verifications for KV stores and window stores, because they are used in different tests. Sorry, I should have been clearer on that before.

Reply:
I see, thanks for the explanation.  The suggestion worked